src/dataflow-types/src/types.rs

```diff
@@ -59,8 +59,10 @@ impl PeekResponse {
 /// Various responses that can be communicated about the progress of a TAIL command.
 #[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
 pub enum TailResponse {
+    /// Progress information about the upper frontier.
+    Progress(Timestamp),
     /// Rows that should be returned in order to the client.
-    Rows(Vec<Row>),
+    Rows(Vec<(Timestamp, Diff, Row)>),
```

Comment:
Is the order important here? I could imagine that it is helpful for e.g. sorting, but it is not the conventional `(data, time, diff)` triple. It's fine with a note, but I imagine people elsewhere accidentally writing `for (data, time, diff) in ...` and being wrong.

Reply:
Done